src/generic/listctrl.cpp

```diff
@@ -1851,6 +1851,9 @@ long wxListMainWindow::HitTestLine(size_t line, int x, int y) const
 
 bool wxListMainWindow::IsHighlighted(size_t line) const
 {
+    if ( line == (size_t)-1 )
+        return false;
+
```

Comment:
I'm not sure we want to do this. IMHO using invalid line here is a bug, i.e., if anything, there should be a `wxCHECK_MSG()` here.

Could you please explain why/how does this function end up called with an invalid argument?

Reply:
At this [line](https://github.com/wxWidgets/wxWidgets/blob/master/src/generic/listctrl.cpp#L2584) `m_current` might be uninitialised yet (with my changes of course), and calling `IsHighlighted()` in this case with invalid line will result in a crash. I removed the test from `IsHighlighted()` and changed the indicated line as follows:
`bool oldWasSelected = HasCurrent() && IsHighlighted(m_current); `